uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/TOILRequest.php

```diff
@@ -134,31 +134,21 @@ private static function validateTOILAmountIsValid($params) {
    * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidTOILRequestException
    */
   private static function validateValidTOILAmountNotGreaterThanMaximum($params) {
-    if(empty($params['contact_id']) || empty($params['type_id'])){
+    if (empty($params['contact_id']) || empty($params['type_id'])) {
       return;
     }
 
     $absenceType = AbsenceType::findById($params['type_id']);
     $unlimitedAccrual = empty($absenceType->max_leave_accrual) && $absenceType->max_leave_accrual != 0;
 
     $currentPeriod = AbsencePeriod::getCurrentPeriod();
```

Comment:
This should not be the current period, but the period for the dates of the TOIL Request. The AbsencePeriod BAO has the `getPeriodContainingDates` method which can be used for that.